lib/aws-record/record/item_collection.rb

```diff
@@ -23,6 +23,13 @@ def each(&block)
         end
       end
 
+      def empty?
+        unless @result
+          @result = @client.send(@search_method, @search_params)
```

Comment:
If we're adding this, I'd like to pull this call down into a shared private method, to avoid code duplication issues.